apps/lib/cmp_mock_srv.c

```diff
@@ -231,6 +231,11 @@ static OSSL_CMP_PKISI *process_cert_request(OSSL_CMP_SRV_CTX *srv_ctx,
     *certOut = NULL;
     *chainOut = NULL;
     *caPubs = NULL;
+    if (certReqId != (p10cr == NULL ? 0 : -1)) {
```

Comment:
Should the `0` and `-1` use the defines?
`OSSL_CMP_CERTREQID` and `OSSL_CMP_CERTREQID_NONE` respectively?

Reply:
I was thinking the same, too, but the mock server code in `apps/` does not have access to these defines, 
which are private in `crypto/cmp/cmp_local.h`.

Thinking a little more about the issue, I found a much better solution:
move the check for the right `certReqId` value from the mock server to the core server implementation,
where it can be done once and for all, without bothering all server application instances with this topic.